src/common/math.h

```diff
@@ -119,14 +136,34 @@ inline static bool CmpSecond(const std::pair<float, unsigned> &a,
 // check nan
 bool CheckNAN(double v);
 #else
-template<typename T>
-inline bool CheckNAN(T v) {
-#ifdef _MSC_VER
-  return (_isnan(v) != 0);
+
+// Redefined here to workaround a VC bug that doesn't support overloadng for integer
```

Comment:
Can you explain a little more the need for this? I never had this problem before.

Reply:
@RAMitchell `std::isnan` for integer types is an extension from c++11, while `isnan` is a c function which doesn't support integer.  This PR requires isnan to deal with integer type and tests from Jenkins said something similar to "no suitable overload"...